src/main/java/com/code_intelligence/jazzer/mutation/mutator/collection/MapMutatorFactory.java

```diff
@@ -126,32 +130,101 @@ public void initInPlace(Map<K, V> reference, PseudoRandom prng) {
       }
     }
 
+    private void eraseRandomChunk(Map<K, V> map, PseudoRandom prng, int mapSize) {
+      int upperLimit = Math.max(1, Math.min(mapSize - minSize, mapSize / 2));
+      int chunkSize = upperLimit == 1 ? 1 : prng.closedRange(1, upperLimit);
+      int chunkOffset = prng.closedRange(0, mapSize - chunkSize);
+      List<K> tmpRemStore = new ArrayList<>(chunkSize);
+      map.entrySet().stream().skip(chunkOffset).limit(chunkSize).forEach(entry -> {
+        tmpRemStore.add(entry.getKey());
+      });
+      map.keySet().removeAll(tmpRemStore);
+    }
+
+    private void insertRandomChunk(Map<K, V> map, PseudoRandom prng, int mapSize) {
+      int chunkSize = prng.closedRange(1, Math.min(maxSize - mapSize, mapSize));
+      for (int i = 0; i < chunkSize; i++) {
+        initElement(map, prng);
+      }
+    }
+
+    private void changeRandomChunk(Map<K, V> map, PseudoRandom prng, int mapSize) {
+      int chunkOffset = prng.indexIn(mapSize);
+      int chunkSize =
+          Math.min(prng.closedRange(1, mapSize - chunkOffset), (int) Math.ceil(mapSize / 10.0));
+      int elementsChanged = 0;
+      while (elementsChanged < chunkSize) {
+        map.entrySet()
```

Comment:
This is also potentially quadratic in the map size. You can collect the entries to mutate once in a temporary structure and then mutate them in a loop.